.sandbox/provision/lifecycle-asg/on-create.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+ASG_NAME=test-windows-provision
+AVAILABILITY_ZONE=us-east-2a
+MAX_RETRIES=3
+
+source ./common.sh
+
+# adjust the terminal output settings 
+redirect_output on-create.log
+
+# TODO: need to lock and prevent race condition
+INSTANCE_IDS=$(aws autoscaling describe-auto-scaling-groups --auto-scaling-group-name $ASG_NAME --query "AutoScalingGroups[].Instances[].InstanceId" --output text)
+INSTANCE_ID=$(echo $INSTANCE_IDS | awk '{print $1}')
+
+aws autoscaling detach-instances --instance-ids $INSTANCE_ID --auto-scaling-group-name $ASG_NAME --no-should-decrement-desired-capacity
+aws ec2 create-tags --resources $INSTANCE_ID --tags Key=Sandbox,Value=$SANDBOX_NAME --tags Key=SandboxID,Value=$SANDBOX_ID
+INSTANCE=$(aws ec2 describe-instances --instance-id $INSTANCE_ID --query 'Reservations[0].Instances[0]' | jq '.')
+PASSWORD_DATA=$(aws ec2 get-password-data --instance-id $INSTANCE_ID)
+
+retry_count=0
+while [ $retry_count -lt $MAX_RETRIES ]; do
```

Comment:
Use `[[` for recent shell scripts, e.g. `while [[ $retry_count -lt $MAX_RETRIES ]]; `